lxc/init.go

```diff
@@ -168,11 +168,31 @@ func (c *cmdInit) create(conf *config.Config, args []string) (lxd.InstanceServer
 			return nil, "", err
 		}
 
-		if network.Type == "bridge" {
-			devicesMap[c.flagNetwork] = map[string]string{"type": "nic", "nictype": "bridged", "parent": c.flagNetwork}
```

Comment:
@stgraber is it expected that using the `--network` flag on `lxc init` will cause an additional NIC be added to the instance, in a way that it won't override the `eth0` NIC from the profile.

E.g.

```
lxc init images:ubuntu/focal c1 -n lxdbr0
lxc config show c1 --expanded
...
devices:
  eth0:
    name: eth0
    network: lxdbr0
    type: nic
  lxdbr0:
    nictype: bridged
    parent: lxdbr0
    type: nic
```
```

Reply:
It's not really meant to be a secondary NIC, no. I think it's just a side effect of the naming currently.
We could change the CLI behavior to use `eth0` as the NIC name and set the `name` property also to `eth0`.

This is more likely to do what the user would expect.

Reply:
Thanks, that is done now.